app/liquid_tags/medium_tag.rb

```diff
@@ -22,10 +23,11 @@ def render(_context)
 
   private
 
-  def parse_url_for_medium_article(url)
-    sanitized_article_url = ActionController::Base.helpers.strip_tags(url).strip
+  def parse_url(url)
+    match = pattern_match_for(url, [REGISTRY_REGEXP])
```

Comment:
@djuber @jgaskins @jeremyf

Let me explain my reasoning behind my changes to this `parse_url` method.

In testing, when I passed a wrong URL in, I discovered that the `rescue` block is extremely slow in returning the error message (and I am not sure why). Plus, the page sort of froze during this long execution.

I am concerned that this is poor user-experience.

So I decided to add a more explicit match test (similar to other unified embeds), which gives immediate feedback on the invalid URL.

Please let me know your thoughts. I understand that `MediumArticleRetrievalService` failures must be rescued, but I decided to take the burden of identifying invalid urls off of that service, and leave it to be concerned with missing or deleted articles. 😆

Reply:
> @djuber @jgaskins @jeremyf
> 
> Let me explain my reasoning behind my changes to this `parse_url` method.
> 
> In testing, when I passed a wrong URL in, I discovered that the `rescue` block is extremely slow in returning the error message (and I am not sure why). Plus, the page sort of froze during this long execution.
> 
> I am concerned that this is poor user-experience.
> 
> So I decided to add a more explicit match test (similar to other unified embeds), which gives immediate feedback on the invalid URL.
> 
> Please let me know your thoughts. I understand that `MediumArticleRetrievalService` failures must be rescued, but I decided to take the burden of identifying invalid urls off of that service, and leave it to be concerned with missing or deleted articles. 😆

@msarit Is it possible that it was actually sending the request to Medium and the slowness you were seeing was the latency of that request?

Reply:
Hmmmm you make a good point there! Well, so what do you think of my "workaround"? I feel bad raising 2 StandardErrors in the same method (feel non-DRY). BUT I did not want the latency of the Medium request to affect feedback on a mismatched URL; to me, that issue is simple enough to just shoot back "hey, something wrong with your URL"

Reply:
This looks like a very valid approach.  Part of the oddity of the method is that you're doing two conceptual things:

1) validating a local value
2) fetching a remote value based on the local value

You could decompose this method into more methods, but then it gets chatty and less "scannable".  Also, the method is super compact, so proceed with this implementation.